app/controllers/payment_gateways/stripe_controller.rb

```diff
@@ -68,11 +68,11 @@ def validate_payment_intent
     end
 
     def valid_payment_intent?
-      @valid_payment_intent ||= begin
-        return false unless params["payment_intent"]&.starts_with?("pi_")
-
-        order_and_payment_valid?
-      end
+      @valid_payment_intent ||= if params["payment_intent"]&.starts_with?("pi_")
+                                  order_and_payment_valid?
+                                else
+                                  false
+                                end
```

Comment:
I think this could be expressed even simpler: 
```suggestion
      @valid_payment_intent ||= params["payment_intent"]&.starts_with?("pi_") 
                                && order_and_payment_valid?
```

Reply:
I hope you don't mind, but I'll assume this is ok, and go ahead and apply this to move things forward.